lib/rubocop/cop/rails/uniq_before_pluck.rb

```diff
@@ -0,0 +1,56 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Prefer the use of uniq before pluck instead of after.
+      #
+      # The use of uniq before pluck is preferred because it executes
+      # within the database.
+      #
+      # @example
+      #   # bad
+      #   Model.where(...).pluck(:id).uniq
+      #
+      #   # good
+      #   Model.where(...).uniq.pluck(:id)
+      #
+      class UniqBeforePluck < RuboCop::Cop::Cop
+        MSG = 'Use uniq before pluck'.freeze
+        DOT_UNIQ = '.uniq'.freeze
+        NEWLINE = "\n".freeze
+
+        def on_send(node)
+          receiver, method_name, *_args = *node
+
+          unless method_name == :uniq &&
```

Comment:
You should add a check for a `nil` receiver. Consider cases like `uniq.something`. They are unlikely, but might still happen.